src/Shared/OpportunisticIntern.cs

```diff
@@ -713,15 +713,26 @@ internal void ReportStatistics(string heading)
                 Console.WriteLine("##########Top Rejected Strings: \n{0} ", string.Join("\n==============\n", topRejectedString.ToArray()));
             }
 
+            private bool TryInternHardcodedString<T>(T candidate, string str, ref string interned) where T : IInternable
+            {
+                if (candidate.StartsWithStringByOrdinalComparison(str))
```

Comment:
Should there be a debug assert here that the lengths match too? When I read it I thought "this can't be right" but it's fine because you're checking the length outside the call to this method.

Reply:
Done, thanks!